src/app/dts/dts.component.ts

```diff
@@ -24,27 +38,61 @@ export class DtsComponent {
     { prop: 'createdAt', name: 'Created'}
   ];
 
-  constructor(private dtsService: DtsService) {
+  constructor(private dtsService: DtsService, private formBuilder: FormBuilder) {
+    this.dtsForm = this.formBuilder.group({
+      templateKey: new FormControl()
+    });
+
+    this.rows = this.dtsService.getTemplates('enrollment');
+
+    this.onValueChanges();
   }
 
   /**
-   * Populates the data grid with a list of templates.
-   * @param documentType indicates type of document
+   * Filters templates by the user entered template key
    */
-  getTemplates(documentType?: string): void {
-    this.rows = this.dtsService.getTemplates(documentType);
+  onValueChanges(): void {
+    this.dtsForm.valueChanges.subscribe(val => {
+      this.rows = this.dtsService.getTemplates('enrollment')
+        .pipe(
+          map(templates => templates.filter( (template) => {
+            return template.templateKey.toLowerCase().includes(val.templateKey.toLowerCase());
+        })));
+    });
   }
 
   /**
-   * Handle row click
+   * Handle selection of a template
    * @param rowEvent event data for the activated row
    */
   rowClick(rowEvent): void {
     if (rowEvent.type === 'click') {
-      this.dtsService.getTemplateByKey(rowEvent.row.docType, rowEvent.row.templateKey).subscribe(data => {
-        // TODO: CEMS 2078 - Frontend DTS displays selected template in template editor
-        console.log(data);
-      });
+
+      this.dtsForm.controls['templateKey'].setValue(rowEvent.row.templateKey);
+
+      this.selectedTemplate = new class implements Template {
```

Comment:
Why that syntax? What about this one to replace lines 73 to 87:
```
this.selectedTemplate = {... rowEvent.row} as Template;
```

Reply:
Fixed.  Implemented recommended change.